app/webapp/src/user-login/contexts/platforms/TwitterContext.tsx

```diff
@@ -111,20 +112,20 @@ export const TwitterContext = (props: PropsWithChildren) => {
           refreshConnected();
           setSearchParams(searchParams);
         } else {
-          localStorage.removeItem(LS_TWITTER_CONTEXT_KEY);
-
           const context = JSON.parse(contextStr) as TwitterSignupContext;
 
           if (context.state !== state_param) {
             throw new Error('Unexpected state');
           }
 
+          setLoginStatus(LoginStatus.LoggingIn);
           appFetch<HandleSignupResult>(`/api/auth/${PLATFORM.Twitter}/signup`, {
             ...context,
             code: code_param,
           }).then((result) => {
             if (result && result.ourAccessToken) {
               setOurToken(result.ourAccessToken);
+              localStorage.removeItem(LS_TWITTER_CONTEXT_KEY);
```

Comment:
I moved this to be right after calling `setOurToken` so that at least one of these keys exist in local storage during the login stage, otherwise there is a brief period of the state being set to 'LoggedOff'